### PR TITLE
Execute flow

### PR DESCRIPTION
**Pull request by gmirmz**

### CODE REVIEW

File: src/pages/api/execute/tasks.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export default async (req: NextApiRequest, res: NextApiResponse) => {
     logger.debug("Request details:", { method: req.method, url: req.url });
 
     const session = await getServerSession(req, res, authOptions);
-    if (!session && env.NODE_ENV === "production") {
+    if (!session) {
```
Comment (gmirmz) [inline src/pages/api/execute/tasks.ts:17]: @abhishekkundu25 You asked me to remove this while we were troubleshooting the access-issues. Shoud we keep it like this or has this been commited wrongfully?

File: src/pages/executeFlow/Execute.tsx [hunk 2]
```diff
@@ -130,273 +156,306 @@ function ExecuteFlow({
...
-              </tds-dropdown>
+        <hr className="divider" />
+        <div className={styles.headerContainer}>
+          <h3 className="tds-headline-03" style={{ marginBottom: "16px" }}>
+            Execution flow
+          </h3>
+          <div className={styles.headerContainer__detailsContainer}>
+            <div className="tds-detail-02">
+              Graph name:{" "}
+              <span className="bold"> Placeholder for graph name</span>
```
Comment (gmirmz) [inline src/pages/executeFlow/Execute.tsx:207]: What do we do with graph name, do we have access to this?

File: src/pages/executeFlow/Execute.tsx [hunk 3]
```diff
@@ -130,273 +156,306 @@ function ExecuteFlow({
...
-                placeholder="Placeholder"
-                onInput={(e: { currentTarget: { value: string } }) =>
-                  setCreatingNewParameter({
-                    ...creatingNewParameter,
-                    value: e.currentTarget.value,
-                  })
-                }
-              ></tds-textarea>
+            <div className="tds-detail-02">
+              User: <span className="bold">Placeholder for user </span>
```
Comment (gmirmz) [inline src/pages/executeFlow/Execute.tsx:210]: What do we do with this? Can we get username/name from session?

File: src/pages/executeFlow/Execute.tsx [hunk 4]
```diff
@@ -130,273 +156,306 @@ function ExecuteFlow({
...
+                  <div className={styles.contentContainer__parameterChoice}>
+                    <tds-radio-button
+                      name="rb-example"
+                      value="newParam"
+                      radio-id="create"
+                      onClick={(e) => {
+                        setSelectedExecutionMethod("Create");
+                        setSelectedParameter(null);
+                      }}
+                      checked={selectedExecutionMethod === "Create"}
```
Comment (gmirmz) [inline src/pages/executeFlow/Execute.tsx:242]: The checked value is for initial render I believe, we should be able to skip these and just go with "checked" on the option that we want to have pre-selected

File: src/pages/executeFlow/Execute.tsx [hunk 5]
```diff
@@ -130,273 +156,306 @@ function ExecuteFlow({
...
+                    <tds-radio-button
+                      name="rb-example"
+                      onClick={() => {
+                        parameters.length &&
+                          setSelectedExecutionMethod("Existing");
+                      }}
+                      disabled={!parameters.length}
+                      value="savedParams"
+                      radio-id="execute"
+                      checked={
```
Comment (gmirmz) [inline src/pages/executeFlow/Execute.tsx:256]: The checked value is for initial render I believe, we should be able to skip these and just go with "checked" on the option that we want to have pre-selected

File: src/pages/executeFlow/Execute.tsx [hunk 6]
```diff
@@ -130,273 +156,306 @@ function ExecuteFlow({
...
+                          name="dropdown"
+                          label="Select Parameter Set"
+                          label-position="outside"
+                          placeholder="Placeholder"
+                          size="sm"
+                          multiselect={false}
+                          onTdsChange={(e) => {
+                            selectParameter(e.detail.value);
+                          }}
+                          filter
```
Comment (gmirmz) [inline src/pages/executeFlow/Execute.tsx:323]: What does filter do here?

File: src/pages/ofd/Ofd.tsx [hunk 7]
```diff
@@ -157,6 +157,7 @@ const ForceGraphComponent: React.FC = ({
   }, [deletePressed]);
 
   const saveData = async (data: GraphBody) => {
+    console.log(data);
```
Comment (gmirmz) [inline src/pages/ofd/Ofd.tsx:160]: Remove

File: src/pages/ofd/Ofd.tsx [hunk 8]
```diff
@@ -438,7 +439,12 @@ const ForceGraphComponent: React.FC = ({
           <span
             id="execute-graph"
             className={styles.page__header__action}
-            onClick={() => router.push(`/executeFlow/id/123`)}
+            // TODO - replace hardcoded value
+            onClick={() =>
+              router.push(
+                `/executeFlow/id/e817ce2f-1824-465b-9c53-88126a9b0b02`
+              )
+            }
```
Comment (gmirmz) [inline src/pages/ofd/Ofd.tsx:447]: Hardcoded?

Comment (abhishekkundu25) [reply src/pages/ofd/Ofd.tsx:447]: yes, this should not be hardcoded, fixing it

File: src/pages/ofd/Ofd.tsx (hunk 7 above)
Comment (abhishekkundu25) [reply src/pages/ofd/Ofd.tsx:160]: yes this should be removed

File: src/pages/executeFlow/Execute.tsx (hunk 3 above)
Comment (abhishekkundu25) [reply src/pages/executeFlow/Execute.tsx:211]: I have removed it for now, for both the graph name we need an api, let's add it back later

File: src/pages/executeFlow/Execute.tsx (hunk 4 above)
Comment (abhishekkundu25) [reply src/pages/executeFlow/Execute.tsx:243]: without this it programmatically changing the radio button selection was not possible, example when we delete but there is not other parameters in the list, so have to make the selection to create new parameter

File: src/pages/api/execute/tasks.ts (hunk 1 above)
Comment (abhishekkundu25) [reply src/pages/api/execute/tasks.ts:17]: yes we can keep it like this, env.NODE_ENV === "production" is a bit problamatic for deployment and build, removed from other places as well

File: src/pages/executeFlow/Execute.tsx (hunk 6 above)
Comment (abhishekkundu25) [reply src/pages/executeFlow/Execute.tsx]: removed

**Review by abhishekkundu25 — APPROVED** [review]

**Pull request merged**